src/phoenix/schemas/core.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from phoenix.db import models
+from sqlalchemy import case, func, select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+
+@dataclass
+class Dataset:
+    id: int
+    name: str
+    description: Optional[str]
+    created_at: datetime
+    updated_at: datetime
+    metadata: Dict[str, Any]
+
+    @classmethod
+    def from_model(cls, model: models.Dataset) -> "Dataset":
+        return cls(
+            id=model.id,
+            name=model.name,
+            description=model.description,
+            created_at=model.created_at,
+            updated_at=model.updated_at,
+            metadata=model.metadata_,
+        )
+
+    async def count_active_examples(self, session: AsyncSession) -> int:
+        result = await session.execute(
+            select(
+                func.sum(
+                    case(
+                        (models.DatasetExampleRevision.revision_kind == "CREATE", 1),
+                        (models.DatasetExampleRevision.revision_kind == "DELETE", -1),
+                        else_=0,
+                    )
+                )
+            )
+            .select_from(models.DatasetExampleRevision)
+            .join(
+                models.DatasetExample,
+                models.DatasetExample.id == models.DatasetExampleRevision.dataset_example_id,
```

Comment:
```suggestion
                onclause=models.DatasetExample.id == models.DatasetExampleRevision.dataset_example_id,
```

Reply:
may I as how this is different?

Reply:
Just a suggestion for making it more explicit.